src/Cake.Npm/AddUser/NpmAddUserSettingsExtensions.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace Cake.Npm.AddUser
+{
+    using System;
+
+    /// <summary>
+    /// Extensions for <see cref="NpmAddUserSettings"/>.
+    /// </summary>
+    public static class NpmAddUserSettingsExtensions
+    {
+        /// <summary>
+        /// Instructs npm to add user for the given registry.
+        /// </summary>
+        /// <param name="settings">The settings.</param>
+        /// <param name="registry">The registry for the user.</param>
+        /// <returns>The <paramref name="settings"/> instance with <see cref="NpmAddUserSettings.Registry"/> set to <paramref name="registry" />.</returns>
+        public static NpmAddUserSettings FromRegistry(this NpmAddUserSettings settings, Uri registry)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException(nameof(settings));
+            }
+
+            if (registry == null)
+            {
+                throw new ArgumentNullException(nameof(registry));
+            }
+
+            settings.Registry = registry;
+            return settings;
+        }
+
+        /// <summary>
+        /// Sets scope for credentials.
+        /// </summary>
+        /// <param name="settings">The settings.</param>
+        /// <param name="scope">Scope of credentials.</param>
+        /// <returns>The <paramref name="settings"/> instance with <paramref name="scope"/> added to <see cref="NpmAddUserSettings.Scope"/>.</returns>
+        public static NpmAddUserSettings AddScope(this NpmAddUserSettings settings, string scope)
```

Comment:
As this method is intended to be called through method chaining,I would prefer `ForScope` to make it more fluent style